framework/source/class/qx/event/handler/GestureCore.js

```diff
@@ -682,27 +718,125 @@ qx.Bootstrap.define("qx.event.handler.GestureCore", {
      * Fires a roll event.
      *
      * @param domEvent {Event} DOM event
+     * @param target {Element} event target
+     * @param rollFactor {Integer} the roll factor to apply
+     */
+    __fireRollEvent: function (domEvent, target, rollFactor) {
+      domEvent.delta = {
+        x: qx.util.Wheel.getDelta(domEvent, "x") * rollFactor,
+        y: qx.util.Wheel.getDelta(domEvent, "y") * rollFactor
+      };
+      domEvent.delta.axis = Math.abs(domEvent.delta.x / domEvent.delta.y) < 1 ? "y" : "x";
+      domEvent.pointerType = "wheel";
+      this._fireEvent(domEvent, "roll", domEvent.target || target);
+    },
+
+    /**
+     * Triggers the adaptative roll scrolling.
+     *
+     * @param target {Element} event target
+     */
+    __performAdaptativeRollScrolling: function (target) {
+      var rollFactor = qx.event.handler.GestureCore.WHEEL_ROLL_FACTOR;
+      if (qx.util.Wheel.IS_TOUCHPAD) {
+        // The domEvent was generated by a touchpad
+        rollFactor = qx.event.handler.GestureCore.TOUCHPAD_ROLL_FACTOR;
+      }
+      this.__lastRollEventTime = new Date().getTime();
+      var reLength = this.__rollEvents.length;
+      for (var i = 0; i < reLength; i++) {
+        var domEvent = this.__rollEvents[i];
+        this.__fireRollEvent(domEvent, target, rollFactor);
+      }
+      this.__rollEvents = [];
+    },
+
+    /**
+     * Ends touch pad detection process.
+     */
+    __endTouchPadDetection: function () {
+      if (this.__rollEvents.length > qx.event.handler.GestureCore.TOUCHPAD_WHEEL_EVENTS_THRESHOLD) {
+        qx.util.Wheel.IS_TOUCHPAD = true;
+      } else {
+        qx.util.Wheel.IS_TOUCHPAD = false;
+      }
+      if (qx.core.Environment.get("qx.debug.touchpad.detection")) {
+        qx.log.Logger.debug(this, "IS_TOUCHPAD : " + qx.util.Wheel.IS_TOUCHPAD);
```

Comment:
I hadn't noticed that. No, the only alternative is changing the superclass to qx.core.Object, but this is too low-level of a class to be considering that at this point. The possibility of wide-ranging bugs is too high. Leave it as it is.